src/main/java/fi/helsinki/cs/tmc/core/communication/TmcApi.java

```diff
@@ -167,13 +169,14 @@ public Course getCourseFromString(String jsonString) {
         Gson mapper = new Gson();
         Course course = mapper.fromJson(courseJson.getAsJsonObject("course"), Course.class);
 
+        if (course == null) {
+            return Optional.absent();
+        }
+
```

Comment:
Can we replace this `if` with  [`Optional.fromNullable(T)`](http://docs.guava-libraries.googlecode.com/git/javadoc/com/google/common/base/Optional.html#fromNullable(T)) ?